ape_alchemy/provider.py

```diff
@@ -132,23 +138,33 @@ def get_virtual_machine_error(self, exception: Exception) -> VirtualMachineError
 
         return VirtualMachineError(message=message)
 
-    def _make_request(self, endpoint: str, parameters: list) -> Any:
-        try:
-            return super()._make_request(endpoint, parameters)
-        except HTTPError as err:
-            response_data = err.response.json()
-            if "error" not in response_data:
-                raise AlchemyProviderError(str(err)) from err
-
-            error_data = response_data["error"]
-            message = (
-                error_data.get("message", str(error_data))
-                if isinstance(error_data, dict)
-                else error_data
-            )
-            cls = (
-                AlchemyFeatureNotAvailable
-                if "is not available" in message
-                else AlchemyProviderError
-            )
-            raise cls(message) from err
+    def _make_request(self, endpoint: str, parameters: list,min_retry_delay:int = 1000, retry_backoff_factor:int = 2,
+        max_retry_delay:int = 30000, max_retries:int = 3, retry_jitter:int = 250) -> Any:
+        
+        for i in range(0,max_retries):
+            try:
+                return super()._make_request(endpoint, parameters)
+            except HTTPError as err:
+                response_data = err.response.json()
+                if "error" not in response_data:
+                    raise AlchemyProviderError(str(err)) from err
+
+                error_data = response_data["error"]
+                message = (
+                    error_data.get("message", str(error_data))
+                    if isinstance(error_data, dict)
+                    else error_data
+                )
+                if "exceeded its compute units" in message:
+                    retry_interval = min(max_retry_delay, min_retry_delay * retry_backoff_factor ** i)
+                    print(f"Alchemy compute units exceeded, retrying #{i} in {retry_interval} ms")
```

Comment:
Would be great to convert this to an INFO log

Reply:
just replaced it with logger.info() call. don't know if anything else is needed.